### PR TITLE
Get current CPU number dynamically Insted of hardcode CPU cores.

### PR DESCRIPTION
**Pull request by rprajapx**
Change-Id: I2721749c05a9b9255956a97cc751b3ff4e9edb82
Tracked-On: OAM-97437
Signed-off-by: Kant, Rohit <rohitx.kant@intel.com>

### CODE REVIEW

**Comment by rprajapx** [comment]
Hi @sgnanase 
Please review PR.

File: thermal/Thermal.cpp [hunk 1]
```diff
@@ -55,7 +56,11 @@ static const char *CPU_LABEL[] = {"CPU0",
                                   "CPU8",
                                   "CPU9",
                                   "CPU10",
-                                  "CPU11"};
+                                  "CPU11",
```
Comment (prabhatchand) [inline thermal/Thermal.cpp:59]: currently, are we using more than 12 CPU core for any of the platform?

File: thermal/Thermal.cpp [hunk 2]
```diff
@@ -258,14 +263,23 @@ static int thermal_get_cpu_usages(CpuUsage *list)
             online = 1;
         }
 
+	//Current Max size supported is 16 cores, for more CPU add check
+	int length = sizeof(CPU_LABEL) / sizeof(CPU_LABEL[0]);
```
Comment (prabhatchand) [inline thermal/Thermal.cpp:267]: Declare length variable in the starting of function where other variables are declared.

File: thermal/Thermal.cpp [hunk 3]
```diff
@@ -258,14 +263,23 @@ static int thermal_get_cpu_usages(CpuUsage *list)
...
+	if(j < length){
         list[size] = (CpuUsage) {
             .name = CPU_LABEL[size],
             .active = active,
             .total = total,
             .isOnline = online
         };
+	    size++;
+	} else {
+	    ALOGE("%s: Current CPU core support exceeds Max CPU core support  ", __func__);
```
Comment (prabhatchand) [inline thermal/Thermal.cpp:277]: Don't we need free(line) here?

File: thermal/Thermal.cpp (hunk 1 above)
Comment (rramakax) [reply thermal/Thermal.cpp:59]: Hi This is for the future. 
As I got informed by Fonn Swee Yee, that intel will support 16 cores. 
I re-verified with JaiKrishna nellamapudi. So added support till 16 CPU cores.

Comment (prabhatchand) [reply thermal/Thermal.cpp:59]: No problem extending this to 16.
But it's a good idea to change this when you really have such platform where you can verify this.

**Review by prabhatchand — APPROVED** [review]
Looks good to me.

**Review by balajim001 — APPROVED** [review]
LGTM

**Pull request merged**